Dockerfile

```diff
@@ -119,6 +119,54 @@ RUN apt-get update && \
 # source underlay for shell
 RUN echo 'source "$UNDERLAY_WS/install/setup.bash"' >> /etc/bash.bashrc
 
+# multi-stage for visualizing
+FROM dever AS visualizer
+
+# install foxglove dependacies
+RUN echo "deb https://dl.cloudsmith.io/public/caddy/stable/deb/debian any-version main" > /etc/apt/sources.list.d/caddy-stable.list
```

Comment:
Maybe we should have a separate docker container in .devcontainer for this? I'd like to keep the root docker file as straight forward as possible for use as the base of other people's systems.

Reply:
See: https://github.com/ros-planning/navigation2/pull/3523#issuecomment-1500124785